### PR TITLE
Optimized carousel.js toggle faders

### PR DESCRIPTION
**Pull request by AaronAtkinson**
Added new options param for buildCarousel function "fadersDisabled" which prevents faders & arrows from being added.
Refactored carousel.js so that:

- Fader elements are not added when the new option is used.
- Trigger elements and intersection observers are only added when needed (not when inifinity scroll is enabled and not if faders are disabled)
- Trigger intersection observer logic toggling faders is in it's own dedicated functions so it can be unit tested.

Resolves: [MWPW-139892](https://jira.corp.adobe.com/browse/MWPW-139892)

Test URLs:
- Before: https://stage--express--adobecom.hlx.page/express/?martech=off
- After: https://refactor-carousel--express--wbstry.hlx.page/express/?martech=off

### CODE REVIEW

**Review by redvelvet511 — APPROVED** [review]

**Review by redvelvet511 — APPROVED** [review]

File: express/blocks/shared/carousel.js [hunk 1]
```diff
@@ -5,106 +5,106 @@ function correctCenterAlignment(plat) {
...
-      if (entry.target === rightTrigger) {
-        if (entry.isIntersecting) {
-          rightControl.classList.add('arrow-hidden');
-          platform.classList.remove('right-fader');
-        } else {
-          rightControl.classList.remove('arrow-hidden');
-          platform.classList.add('right-fader');
-        }
+      if (entry.target.classList.contains('carousel-left-trigger')) {
+        if (entry.isIntersecting) hideFader(leftControl);
```
Comment (vhargrave) [inline express/blocks/shared/carousel.js:53]: this code would also look nicer with that toggleMethod.

File: express/blocks/shared/carousel.js [hunk 2]
```diff
@@ -5,106 +5,106 @@ function correctCenterAlignment(plat) {
...
 
-  faderLeft.addEventListener('click', () => {
-    const increment = Math.max((platform.offsetWidth / 4) * 3, 300);
-    moveCarousel(increment);
-  });
-  faderRight.addEventListener('click', () => {
-    const increment = Math.max((platform.offsetWidth / 4) * 3, 300);
-    moveCarousel(-increment);
-  });
+  if (!options.fadersDisabled) {
```
Comment (vhargrave) [inline express/blocks/shared/carousel.js:81]: Do you know where this code will be used? 
i'm suddenly a little concerned that we've built something that we don't need 😅

**Comment by vhargrave** [comment]
Merge conflicts also need to be resolved.

File: express/blocks/shared/carousel.js (hunk 1 above)
Comment (AaronAtkinson) [reply express/blocks/shared/carousel.js]: The toggleFader function definitely improved the readability and simplicity here!

File: express/blocks/shared/carousel.js (hunk 2 above)
Comment (AaronAtkinson) [reply express/blocks/shared/carousel.js:74]: This was added for an experiment, but is no longer being used for that, or anywhere else.

Comment (AaronAtkinson) [reply express/blocks/shared/carousel.js]: I removed the option 👍

**Review by vhargrave — CHANGES_REQUESTED** [review]
linters/tests are failing
when I go to our homepage and change to a different tab such as photo, the carousel doesn't get initialized anymore.

**Pull request closed without merging**